Description: Fix how link dependencies are specified
 Several target have their dependencies incorrectly reported in _LDFLAGS
 variables instead of _LIBADD or _LDADD. AS a result, the link flag
 -Wl,--as-needed is misplaced making it useless. This patch sanetizes this
 a little bit.
Author: Nicolas Bourdaud <nicolas.bourdaud@gmail.com>
Origin: vendor
Forwarded: no
Last-Update: 2012-07-28

--- a/src/Makefile-hotplug-sniffer.am
+++ b/src/Makefile-hotplug-sniffer.am
@@ -15,7 +15,7 @@ cinnamon_hotplug_sniffer_CFLAGS =		\
 	$(CINNAMON_HOTPLUG_SNIFFER_CFLAGS)		\
 	$(NULL)
 
-cinnamon_hotplug_sniffer_LDFLAGS =		\
+cinnamon_hotplug_sniffer_LDADD =		\
 	$(CINNAMON_HOTPLUG_SNIFFER_LIBS)		\
 	$(NULL)
 
diff --git a/src/Makefile-st.am b/src/Makefile-st.am
index e9ad239..3167bb2 100644
--- a/src/Makefile-st.am
+++ b/src/Makefile-st.am
@@ -175,7 +175,6 @@ libst_1_0_la_SOURCES =				\
 	$(st_built_sources)			\
 	$(NULL)
 libst_1_0_la_CPPFLAGS = $(st_cflags)
-libst_1_0_la_LDFLAGS = $(LDADD)
 
 noinst_PROGRAMS += test-theme
 
diff --git a/src/Makefile-tray.am b/src/Makefile-tray.am
index 1377e0c..2d32c9f 100644
--- a/src/Makefile-tray.am
+++ b/src/Makefile-tray.am
@@ -49,7 +49,6 @@ libtray_la_SOURCES =		\
 	$(tray_built_sources)	\
 	$(NULL)
 libtray_la_CPPFLAGS = $(tray_cflags)
-libtray_la_LDFLAGS = $(LDADD)
 
 CLEANFILES += $(TRAY_STAMP_FILES) $(BUILT_SOURCES)
 
